orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -6061,3 +6063,45 @@ def generate_inputs(batch_size, max_seq_length, vocab_size):
     assert "ShrunkenGather" in [node.op_type for node in training_model.graph.node]
     assert "PadAndUnflatten" in [node.op_type for node in training_model.graph.node]
     del os.environ["ORTMODULE_ENABLE_EMBEDDING_SPARSE_OPTIMIZER"]
+
+
+def test_cache_exported_model():
+    os.environ["ORTMODULE_CACHE_DIR"] = "cache_dir"
+
+    class Net(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.fc = torch.nn.Linear(10, 1)
+
+        def forward(self, x):
+            x = x.view(x.shape[0], -1)
+            x = torch.nn.functional.relu(self.fc(x))
+            return x
+
+    data = torch.randn(1, 10)
+
+    # first time seeing the model, architecture should be cached under ORTMODULE_CACHE_DIR
+    model_pre_cache = Net()
+    model_pre_cache = ORTModule(model_pre_cache, DebugOptions(log_level=LogLevel.INFO))
+
+    pre_cache_start = time.time()
+    _ = model_pre_cache(data)
+    pre_cache_duration = time.time() - pre_cache_start
+
+    root_dir = Path(__file__).resolve().parent
+    cache_dir = root_dir / os.environ["ORTMODULE_CACHE_DIR"]
+
+    assert len(os.listdir(cache_dir)) == 1
+
+    # second time seeing the model, architecture should be loaded from ORTMODULE_CACHE_DIR
+    model_post_cache = Net()
+    model_post_cache = ORTModule(model_post_cache, DebugOptions(log_level=LogLevel.INFO))
+
+    post_cache_start = time.time()
+    _ = model_post_cache(data)
+    post_cache_duration = time.time() - post_cache_start
+
+    assert post_cache_duration < pre_cache_duration
+
+    shutil.rmtree(cache_dir)
```

Comment:
Instead of creating and deleting the dir, can you instead use tempfile temporary directory (some of the other tests here do that I believe).